tests/integration/aws/general/package/tests.js

```diff
@@ -0,0 +1,43 @@
+'use strict';
+
+const path = require('path');
+const expect = require('chai').expect;
+const execSync = require('child_process').execSync;
+const AWS = require('aws-sdk');
+const EOL = require('os').EOL;
+const fs = require('fs');
+
+const CF = new AWS.CloudFormation({ region: 'us-east-1' });
+const Utils = require('../../../../utils/index');
+
+describe('AWS - General: Deployment with --noDeploy', function () {
+  this.timeout(0);
+  let serviceName;
+  let deploy;
+
+  before(() => {
+    serviceName = Utils.createTestService('aws-nodejs', path.join(__dirname, 'service'));
+    deploy = execSync(`${Utils.serverlessExec} deploy --noDeploy`);
+  });
+
+  it('should deploy package with --noDeploy flag', () => {
+    const result = new Buffer(deploy, 'base64').toString();
+    const resultLines = result.split(EOL);
+    expect(resultLines[0]).to.equal('Serverless: Packaging service...');
+    expect(resultLines[1]).to.have.string('--noDeploy');
+  });
+
+  it('should have create cloudformation files and functions zip', () => {
+    const deployedFiles = fs.readdirSync(path.join(process.cwd(), '.serverless'));
+    expect(deployedFiles[0]).to.equal('cloudformation-template-create-stack.json');
+    expect(deployedFiles[1]).to.equal('cloudformation-template-update-stack.json');
+    expect(deployedFiles[2]).to.match(/service-[0-9]{13}.zip/);
```

Comment:
When it comes to testing I recommend to rather be very precise and fix a test, before something not obvious slippy through. Otherwise the notion will come up: "Integration Tests are useless"